cli/command/telemetry_utils.go

```diff
@@ -100,6 +101,66 @@ func startCobraCommandTimer(mp metric.MeterProvider, attrs []attribute.KeyValue)
 	}
 }
 
+// basePluginCommandAttributes returns a slice of attribute.KeyValue to attach to metrics/traces
+func basePluginCommandAttributes(plugincmd *exec.Cmd, streams Streams) []attribute.KeyValue {
+	pluginPath := strings.Split(plugincmd.Path, "-")
+	pluginName := pluginPath[len(pluginPath)-1]
+	return append([]attribute.KeyValue{
+		attribute.String("plugin.name", pluginName),
+	}, stdioAttributes(streams)...)
+}
+
+// wrappedCmd is used to wrap an exec.Cmd in order to instrument the
+// command with otel by using the TimedRun() func
+type wrappedCmd struct {
```

Comment:
Instead of creating a wrapped `exec.Cmd` type, how hard would it be to just have `pluginmanager.PluginRunCommand` return a `func` that starts a timer, runs the command, and ends the timer?

Okay if we end up going with a wrapper, I just try hard to avoid those/introducing more state since now it has a pointer to a cli, etc.

Reply:
i'm not sure that'd work..in the sense that we do things with the `exec.Cmd` that `pluginmanager.PluginRunCommand` returns aside from just running it (see the `tryPluginRun()` func as an example). So i wanted to keep the return value of `InstrumentPluginCommand` something close to what `pluginmanager.PluginRunCommand` currently returns. I guess we could rework all that but i'm not sure it'd be worth the extra work. 

I'm not a fan of the wrapped type here either, but it seemed like a way to keep code changes relatively minimal on the usage side of things, and to allow code that doesn't need the otel bits to continue to function as usual. Happy to change the impl if we can find some simpler way to handle this. The pointer to DockerCli is really only needed to get the `meter provider` and the `startPluginCommandTimer` func atm.

The main idea of this impl was to try and keep the approach as close as possible to the one used by `InstrumentCobraCommand`

Reply:
Gotcha. Let me take a closer look, but if this is indeed the least invasive way that's okay by me. I also don't love having to remember to call a new method `TimedRun`, which is why I'd have rather returned a func or something.